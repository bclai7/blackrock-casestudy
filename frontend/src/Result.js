import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Table from 'react-bootstrap/Table';

function Result(props) {
  const results = props.results;
  const validInput = props.validInput;
  const errorOccurred = props.errorOccurred;
  if(errorOccurred) {
    return (
        <Row>
            <Col>
                <div className='errorOccured'>An error has occurred</div>
            </Col>
        </Row>
    );
  }
  if(validInput) {
    return (
        <div className="Result">
            <div className='resultHeader sectionHeader'>
                <Row>
                    <Col>
                        Results
                    </Col>
                </Row>
          </div>
    
          <div className='resultSection'>
                <Row>
                    <Col>
                    <Table bordered hover>
                        <thead className='tableHeader'>
                            <tr>
                                <th>N</th>
                                <th>X</th>
                                <th>Y</th>
                                <th>Z</th>
                            </tr>
                        </thead>
                        <tbody>
                            {results?.map((result) => (
                                <tr key={result.n}>
                                    <td>{result.n}</td>
                                    <td>{result.x}</td>
                                    <td>{result.y}</td>
                                    <td>{result.z}</td>
                                </tr>
                            ))}
                        </tbody>
                    </Table>
                    </Col>
                </Row>
          </div>
        </div>
      );
  }
  else {
    return (
        <Row>
            <Col>
                <div className='invalidInput'>Invalid Input</div>
            </Col>
        </Row>
    );
  }
}

export default Result;