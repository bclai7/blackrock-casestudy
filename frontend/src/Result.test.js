import { render, screen, cleanup } from '@testing-library/react';
import Result from './Result';
import renderer from 'react-test-renderer';

test('renders Result component', () => {
  render(<Result results = { [] } validInput = {true} errorOccurred={false} />);
  const resultHeader = screen.getByTestId('resultHeader');
  const resultSection = screen.getByTestId('resultSection');
  const resultTable = screen.getByTestId('resultTable');
  expect(resultHeader).toBeInTheDocument();
  expect(resultSection).toBeInTheDocument();
  expect(resultTable).toBeInTheDocument();
});

test('invalid input', () => {
    render(<Result results = { [] } validInput = {false} errorOccurred={false} />);
    const invalidInput = screen.getByTestId('invalidInput');
    expect(invalidInput).toBeInTheDocument();
    expect(invalidInput).toHaveTextContent('Invalid Input');
    
});

test('error occurred', () => {
    render(<Result results = { [] } validInput = {true} errorOccurred={true} />);
    const errorOccurred = screen.getByTestId('errorOccurred');
    expect(errorOccurred).toBeInTheDocument();
    expect(errorOccurred).toHaveTextContent('An error has occurred');
    
});

test('successful result snapshot', () => {
    const snap = renderer.create(<Result results = { [] } validInput = {true} errorOccurred={false} />).toJSON();
    expect(snap).toMatchSnapshot();
});

test('invalid input result snapshot', () => {
    const snap = renderer.create(<Result results = { [] } validInput = {false} errorOccurred={false} />).toJSON();
    expect(snap).toMatchSnapshot();
});

test('error occureed result snapshot', () => {
    const snap = renderer.create(<Result results = { [] } validInput = {true} errorOccurred={true} />).toJSON();
    expect(snap).toMatchSnapshot();
});