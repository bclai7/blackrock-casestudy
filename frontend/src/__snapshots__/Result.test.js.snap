// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error occureed result snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="errorOccured"
      data-testid="errorOccurred"
    >
      An error has occurred
    </div>
  </div>
</div>
`;

exports[`invalid input result snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="invalidInput"
      data-testid="invalidInput"
    >
      Invalid Input
    </div>
  </div>
</div>
`;

exports[`successful result snapshot 1`] = `
<div
  className="Result"
  data-testid="resultHeader"
>
  <div
    className="resultHeader sectionHeader"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        Results
      </div>
    </div>
  </div>
  <div
    className="resultSection"
    data-testid="resultSection"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <table
          className="table table-bordered table-hover"
          data-testid="resultTable"
        >
          <thead
            className="tableHeader"
          >
            <tr>
              <th>
                N
              </th>
              <th>
                X
              </th>
              <th>
                Y
              </th>
              <th>
                Z
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
